CredentialProvider.Microsoft/CredentialProviders/Vsts/VstsCredentialProvider.cs

```diff
@@ -81,8 +78,35 @@ public override async Task<GetAuthenticationCredentialsResponse> HandleRequestAs
                         responseCode: MessageResponseCode.Error);
                 }
 
-                var sessionTokenClient = new VstsSessionTokenClient(request.Uri, bearerToken, authUtil);
-                var sessionToken = await sessionTokenClient.CreateSessionTokenAsync(endTime, cancellationToken);
+                // Allow the user to choose their token type
+                // If they don't and interactive auth was required, then prefer a PAT so we can safely default to a much longer validity period
+                VstsTokenType tokenType = EnvUtil.GetVstsTokenType() ??
+                    (bearerTokenResult.ObtainedInteractively
```

Comment:
@satbai or @zarenner  - Any recollection on why this pivoted on interactive vs non-interactive?

Reply:
@johnterickson because if we could obtain the token non-interactively, we assume we could obtain the token again non-interactively, so it's safe to have a very short validity period without annoying the user with re-auths, and if we have a very short validity period, we can just request a self-describing token that's a tiny bit more efficient on the server and doesn't clutter up the user's PAT UI

If we had to prompt the user, then we wanted to request a longer lifetime so we don't annoy them with repeated prompts, for which we wanted to obtain a PAT, which shows up in the UI and can be revoked or extended by the user as usual for PATs